src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -484,6 +964,118 @@ get_services_properties(sd_bus_message *m, const char *path)
 
             r = sd_bus_message_exit_container(m);
             SOL_INT_CHECK(r, < 0, r);
+        }  else if (streq(str, "Security")) {
+            const char *security = NULL;
+
+            sol_ptr_vector_clear(&service->security);
+
+            r = sd_bus_message_enter_container(m, 'a', "s");
+            SOL_INT_CHECK(r, < 0, r);
+
+            while (sd_bus_message_read_basic(m, 's', &security) > 0) {
+                r = sol_ptr_vector_append(&service->security, security);
+                SOL_INT_CHECK(r, < 0, r);
```

Comment:
You should exit the container and you will leave the nameservers. I think you should iterate over the list the clear all the elements and them clear the list itself.